client/devdocs/design/index.jsx

```diff
@@ -83,44 +97,44 @@ export default React.createClass( {
 					</SearchCard>
 				}
 				<Collection component={ this.props.component } filter={ this.state.filter }>
-					<Accordions />
```

Comment:
I have been asking myself the same question from the beginning :)
Initially I thought that I could define a property on the example component, say `documents`, and expose the `displayName` of the documented component. E.g.:
```javascript
Buttons.documents = Button.displayName || Button.name;
```
That would allow me to do something like this in `client/devdocs/design/search-collection.jsx`:
```javascript
examples = React.Children.map( this.props.children, ( example ) => {
	const newProps = Object.assign( {}, example.props, {
		componentUsageStats: componentsUsageStats[ camelCase( example.documents ) ] || {}
	} );

	return (
		<Hider hide={ this.shouldWeHide( example ) } key={ 'example-' + example.type.displayName }>
			{
				React.cloneElement(
					example,
					newProps
				)
			}
		</Hider>
	);
} );
```
The problem here is that `displayName` is meant to be used for debugging/tooling usually and it is accessible via `component.type.displayName` also any other property like `documents` is ignored and not exposed so I wasn't able to do the thing above – maybe there is a way though and you know it? :)

Maybe another option is to pass the `componentsUsageStats` (or simply the `state` object?) down to each single example and then use a [selector](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#selectors).
I am not familiar with `selectors` so I should dig into this topic but sounds like a reasonable approach to me, thoughts?